### PR TITLE
User route

### PR DESCRIPTION
**Pull request by Miminiverse**
This PR contains:
1/ Logout route
2/ Protected route, user will be redirected accordingly if they are not authenticated
3/ Store user info in UseContext and display username in Navbar when user log in

*** Issue *** 
When user first log in and navigate to Chat page, the user info is not available. I will have to refresh the page so that I will receive user info. This issue will be in Jira: 

https://bbpracticum.atlassian.net/browse/CP2-25?atlOrigin=eyJpIjoiYWU0MTlmMzdjYWVlNDFiYmI1NGRjNzM0NDkwNWU2YTkiLCJwIjoiaiJ9below:

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,17 +1,33 @@
...
+
+  useEffect(() => {
+    if (localStorage.getItem('token')) {
+      navigate("/chat");
+    } else {
+      navigate('/')
+    }
+  }, []);
+  
+
```
Comment (Said-laasri) [inline src/App.tsx:24]: when user is already sign in and token is in local storage the user is redirected automatically to the chat app 
Great job :+1:

File: src/navbar/NavBar.tsx [hunk 2]
```diff
@@ -1,11 +1,42 @@
...
+             Logout
+           </button>
+           : null  
+            }
+            </li>
+            <li className="items-center" >
+                {user && user.userName ? <p> Hello, {user.userName} </p> : ""}
+            </li>
+          </ul>
         </nav>
```
Comment (Said-laasri) [inline src/navbar/NavBar.tsx:37]: on Log-out the user is taken the login page
:100:

File: src/context/user-context.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+            setUser(loggedInUser)
+        }
+    }
+    
+
+    return (
+        <UserContext.Provider value={[user, setUser]}>
+            {children}
+        </UserContext.Provider>
+    )
```
Comment (Said-laasri) [inline src/context/user-context.tsx:30]: on sign-in user context get information from decoding token :1st_place_medal: 

The user name is displayed on the chat app page

**Review by Said-laasri — APPROVED** [review]
**Hi @Miminiverse**  :100: 



Greta job implementation:
Secure routing :+1: 
use context :1234: 
Token decoder :100: 

HAPPY CODING!! :1st_place_medal:

File: src/entryPoint/Login.tsx [hunk 4]
```diff
@@ -70,8 +75,9 @@ const navigate = useNavigate()
           showNotification('User signed in');
```
Comment (royemosby) [inline src/entryPoint/Login.tsx:76]: We may at one point want to look at generalizing showNotification. I could see other notifications inside the UI being useful. The developer experience would improved by using a single mechanism to handle a variety of alerts at a higher level in the component tree.

File: src/index.tsx [hunk 5]
```diff
@@ -4,12 +4,15 @@ import App from './App';
...
 import './index.css'
+import {UserContextProvider} from "./context/user-context"
 
 const root = ReactDOM.createRoot(document.getElementById('root'));
 root.render(
+    <UserContextProvider>
     <BrowserRouter>
        <App />
-    </BrowserRouter>,
+    </BrowserRouter>
```
Comment (royemosby) [inline src/index.tsx:14]: Minor nit- watch indentation

**Review by royemosby — APPROVED** [review]
I left a couple minor comments but everything's good to go!

**Pull request merged**